esmvalcore/preprocessor/_bias.py

```diff
@@ -0,0 +1,114 @@
+"""Preprocessor functions to calculate biases from data."""
+import logging
+
+import dask.array as da
+import iris.cube
+
+from ._io import concatenate
+
+logger = logging.getLogger(__name__)
+
+
+def bias(products, bias_type='absolute', denominator_mask_threshold=1e-3,
+         keep_reference_dataset=False):
+    """Calculate biases.
+
+    Notes
+    -----
+    This preprocessor requires a reference dataset. For this, exactly one input
+    dataset needs to have the facet ``reference_for_bias: true`` defined in the
+    recipe. In addition, all input datasets need to have identical dimensional
+    coordinates. This can for example be ensured with the preprocessors
+    :func:`esmvalcore.preprocessor.regrid` and/or
+    :func:`esmvalcore.preprocessor.regrid_time`.
+
+    Parameters
+    ----------
+    products: set of esmvalcore.preprocessor.PreprocessorFile
+        Input datasets. Exactly one datasets needs the facet
+        ``reference_for_bias: true``.
+    bias_type: str, optional (default: 'absolute')
+        Bias type that is calculated. Must be one of ``'absolute'`` (dataset -
+        ref) or ``'relative'`` ((dataset - ref) / ref).
+    denominator_mask_threshold: float, optional (default: 1e-3)
+        Threshold to mask values close to zero in the denominator (i.e., the
+        reference dataset) during the calculation of relative biases. All
+        values in the reference dataset with absolute value less than the given
+        threshold are masked out. This setting is ignored when ``bias_type`` is
+        set to ``'absolute'``.
+    keep_reference_dataset: bool, optional (default: False)
+        If ``True``, keep the reference dataset in the output. If ``False``,
+        drop the reference dataset.
+
+    Returns
+    -------
+    set of esmvalcore.preprocessor.PreprocessorFile
+        Output datasets.
+
+    Raises
+    ------
+    ValueError
+        Not exactly one input datasets contains the facet
+        ``reference_for_bias: true``; ``bias_type`` is not one of
+        ``'absolute'`` or ``'relative'``.
+
+    """
+    # Get reference product
+    reference_product = []
+    for product in products:
+        if product.attributes.get('reference_for_bias', False):
+            reference_product.append(product)
+    if len(reference_product) != 1:
```

Comment:
I think it's best you put a check for multiple `reference_for_bias` entries in the recipe in `_recipe_checks` - this will have failed only after a costly regrid operation was performed

Reply:
Will do!